pkg/render/logstorage.go

```diff
@@ -734,24 +745,22 @@ func (es elasticsearchComponent) nodeSetTemplate(pvcTemplate corev1.PersistentVo
 		"node.ingest":                 "true",
 		"cluster.max_shards_per_node": 10000,
 	}
-
-	switch es.authentication.(type) {
-	case OIDCAuthentication:
-		oidcAuth := es.authentication.(OIDCAuthentication)
+	if es.dexCfg != nil {
 		config["xpack.security.authc.realms.oidc.oidc1"] = map[string]interface{}{
 			"order":                       1,
-			"rp.client_id":                oidcAuth.ClientID,
+			"rp.client_id":                DexClientId,
 			"rp.response_type":            "code",
-			"rp.redirect_uri":             fmt.Sprintf("https://%s:9443/tigera-kibana/api/security/oidc/callback", oidcAuth.SiteURL),
-			"rp.requested_scopes":         oidcAuth.RequestedScopes,
-			"op.issuer":                   oidcAuth.IssuerURL,
-			"op.authorization_endpoint":   oidcAuth.AuthorizationEndpoint,
-			"op.token_endpoint":           oidcAuth.TokenEndpoint,
-			"op.jwkset_path":              oidcAuth.JWKSetURI,
-			"op.userinfo_endpoint":        oidcAuth.UserInfoEndpoint,
-			"rp.post_logout_redirect_uri": fmt.Sprintf("https://%s:9443/tigera-kibana/logged_out", oidcAuth.SiteURL),
-			"claims.principal":            oidcAuth.UsernameClaim,
-			"claims.groups":               oidcAuth.GroupsClaim,
+			"rp.redirect_uri":             fmt.Sprintf("%s/tigera-kibana/api/security/oidc/callback", es.dexCfg.ManagerURI()),
+			"rp.requested_scopes":         es.dexCfg.RequestedScopes(),
+			"rp.post_logout_redirect_uri": fmt.Sprintf("%s/tigera-kibana/logged_out", es.dexCfg.ManagerURI()),
```

Comment:
I think that most of these ones that are using fmt.Sprintf or are hard coded should be returned by the dexCfg. I think it might make sense to have functions like TokenEndpoint(), UserInfoEndpoint() and Authorization() endpoints so that anything else that requires this information wouldn't have to know to hard code it or how to fmt.Sprintf it, it just gets it from the config. This also helps if something changes, then you wouldn't need to update a bunch of different places.